initrd/etc/luks-functions

```diff
@@ -295,153 +309,267 @@ select_luks_container()
-			cryptsetup open $LUKS test --key-file /tmp/luks_current_Disk_Recovery_Key_passphrase
+			echo -n "$luks_current_Disk_Recovery_Key_passphrase" > /tmp/secret/luks_current_Disk_Recovery_Key_passphrase
 		else
-			echo -n "$luks_current_Disk_Recovery_Key_passphrase" >/tmp/luks_current_Disk_Recovery_Key_passphrase
-			warn "Testing opening "$LUKS" LUKS encrypted drive content with the current LUKS Disk Recovery Key passphrase..."
-			cryptsetup open $LUKS test --key-file /tmp/luks_current_Disk_Recovery_Key_passphrase
+			echo -n "$luks_current_Disk_Recovery_Key_passphrase" > /tmp/secret/luks_current_Disk_Recovery_Key_passphrase
 		fi
-		#Validate past cryptsetup-reencrypt attempts
-		if [ $? -eq 0 ]; then
-			whiptail --title 'Invalid Actual LUKS Disk Recovery Key passphrase?' --msgbox \
-				"If you previously changed it and do not remember it, you will have to\n reinstall the OS from a an external drive.\n\nTo do so, place the ISO file and its signature file on root of an\n external drive, and select Options-> Boot from USB \n\nHit Enter to retry." 30 60
-			shred -n 10 -z -u /tmp/luks_current_Disk_Recovery_Key_passphrase 2>/dev/null
-			#unsetting luks_current_Disk_Recovery_Key_passphrase so we prompt for it again LUKS Disk Recovery Key passphrase prompt on next round
-			unset luks_current_Disk_Recovery_Key_passphrase
-			#remove "known good" selected LUKS container so that next pass asks again user to select LUKS container.
-			#maybe the container was not the right one
-			detect_boot_device
-			mount -o remount,rw /boot
-			rm -f /boot/kexec_key_devices.txt
-			mount -o remount,ro /boot
-		else
-			#LuksOpen test was successful. Cleanup should be called only when done
-			#Exporting successfully used passphrase possibly reused by oem-factory-reset
 
-			#We close the volume
-			cryptsetup close test
-			export luks_current_Disk_Recovery_Key_passphrase
+		# test all LUKS containers on same block device as returned by select_luks_container
+		echo -e "\n$PRINTABLE_LUKS: Test unlocking of LUKS encrypted drive content with current LUKS Disk Recovery Key passphrase..."
+
+		# Loop on all LUKS containers on same block device
+		for luks_container in $LUKS; do
+			DEBUG "$luks_container: Test unlocking of LUKS encrypted drive content with current LUKS Disk Recovery Key passphrase..."
+			DO_WITH_DEBUG cryptsetup open --test-passphrase "$luks_container" --key-file /tmp/secret/luks_current_Disk_Recovery_Key_passphrase
+			# Validate past cryptsetup reencrypt attempts
+			if [ $? -ne 0 ]; then
+				# if we have more than one LUKS container and passphrase test unsuccessful, tell user how to change passphrase
+				if [ $(echo $LUKS | wc -w) -gt 1 ]; then
+					#TODO remove this once whiptail_error whiptail_warning can take titles with double quotes
+					#whiptail_warning --title 'tes' --msgbox 'test' 0 80
+					#whiptail_error --title 'error' --msgbox 'error' 0 80
+					#Neither work today. Not related to this PR... Using whiptail without coloring.
+
```

Comment:
What happened when you tried to do this?  We do this elsewhere

Reply:
I tried to pass a title with a expended variable and that failed. Will have to revisit later, this is left to investigate outside of this PR, where technological debt is that the added whiptail calls are not whiptail_warning now, I don't know what I missed.

Reply:
@JonathonHall-Purism I left notes here because I was unsuccessful calling whiptail_warn and whiptail_error with a varibale expanding in --title, which works in messagebox.

Reply:
These comments here don't really help with that and I don't see any other purpose to them :shrug: 

Variable expansions are done by the shell before invoking the command (which here is a shell function), there should be no difference between expanding a variable in different parameters.

You didn't accidentally leave it in single quotes right?

Reply:
Was fixed in master, removed and changed code with wuiptail_error where needed under https://github.com/linuxboot/heads/pull/1541/commits/cd6688589076bc932ba323203126e1a3c1f4bd37